vercel-functions/prerender-user-profile-page.func/index.js

```diff
@@ -0,0 +1,55 @@
+import FastBoot from 'fastboot';
+import { replaceAllMetaTags } from './replace-meta-tag.js';
+
+export default async function (request, response) {
+  const { username } = request.query;
+
+  // Check if username query parameter is provided
+  if (!username) {
+    console.error('Missing "username" query parameter');
+    response.redirect('/404');
+
+    return;
+  }
+
+  // Initialize a FastBoot instance
+  const app = new FastBoot({
+    distPath: 'dist',
+    resilient: false,
+
+    // Customize the sandbox globals
+    buildSandboxGlobals(defaultGlobals) {
+      return Object.assign({}, defaultGlobals, {
+        AbortController,
+      });
+    },
+
+    maxSandboxQueueSize: 1,
+  });
+
+  // Visit the user profile page
+  const result = await app.visit(`/users/${username}`);
```

Comment:
It sucks that we have to write this individually for each route - @VasylMarchuk wonder if there's a way to make this work for "any" route? 

For example, maybe we could use a single `/prerender.func` as a serverless function, and have that take `?path=/users/abcd` as a query param?

Reply:
I loved this idea [at first], been thinking about it for a couple hours.

Wondering if it's possible to redirect multiple routes to a function, for example:

```
/users/:suffix --> /prerender-ember-route?prefix=/users/
/concepts/:suffix --> /prerender-ember-route?prefix=/concepts/
```

and have the function receive `prefix` and `suffix` as query parameters.

I'm not sure if vercel's routing can do that (the custom `?prefix=...` part in the `destination`), will check if it can.

Reply:
This ^ is the best idea so far, I think, directory structure would look like:

```
functions/
  prerender/
    _prerender_ember_route.func/
    users/
      user.func <-- symlink to `../_prerender_ember_route.func`
    concepts/
      concept.func <-- symlink to `../_prerender_ember_route.func`
```

Direct requests to `/prerender/**` would be redirected to `/404`

Requests to `/users/:id` would be rewritten to `/prerender/users/user?id=...`
Requests to `/concepts/:id` would be rewritten to `/prerender/concept/contest?id=...`

And the function `_prerender_ember_route` would be able to tell the ember path to prerender from the alias:

```
/prerender/users/user/:id --> visit `/users/user/:id`
/prerender/concepts/concept/:id --> visit `/concepts/concept/:id`
```

Reply:
Converted to the symlinks approach